Automattic/wp-calypso
# Plans Step: TS and function form

Related to #

## Proposed Changes

- Transforms the Plans step from the Start framework into TS and function form.
- Files for serving plans-step in both Start and Stepper are now migrated to `stepper/../unified-plans/` and imported from there (and re-exported from Start for other steps). So that's the new home, including styles & tests.
  - Some cleanup will need to happen once Start is fully deprecated/removed - essentially removing the respective `AsyncLoad` and whatever properties are associated. 
  - The previous Plans step in Stepper can similarly be removed in place of this "unified" one. This can happen before Start removal to get everything (the other Stepper flows) aligned over the same step component. We can rename "unified-plans" to just "plans" then.
- Updated tests:
  - kept a bare-bones high-level suite for the plans step (which can be extended further, if need be)
  - adopted the remaining tests from Start on the `buildUpgradeHandler` step action directly

## Why are these changes being made?


This step has been imported and used in Stepper, which is a fully typed system. The code remained in its arcane form, potentially causing unnoticed inconsistencies in behavior/tracking/etc. A quick surface look points to a few properties wrapped from the old signup (Start) not passed from Stepper.

## Testing Instructions



* Need general sanity checks on the Plans step in onboarding flows: 
  * `/start/plans`
  * `/start/onboarding-pm/plans`
  * `/setup/onboarding/plans`
* Confirm buttons work, purchasing a plan
* General sanity check of respective flows overall as the plans step contains back button logic, etc. So back/forth to domains & checkout, etc.

## Pre-merge Checklist
